compiler/noirc_evaluator/src/ssa/ir/instruction/constrain.rs

```diff
@@ -2,6 +2,44 @@ use acvm::{acir::AcirField, FieldElement};
 
 use super::{Binary, BinaryOp, ConstrainError, DataFlowGraph, Instruction, Type, Value, ValueId};
 
+/// Try to simplify this constrain instruction. This function will inspect the inputs to the constraint such that
+/// it acts on variables as early in the DFG as possible.
+pub(super) fn simplify_constrain(
```

Comment:
Was this to try to limit the scope of the clash with the constant folding w/ constraints issue you mentioned?

Reply:
Yes, that's correct.